forge-game/src/main/java/forge/game/ability/effects/UntapAllEffect.java

```diff
@@ -20,22 +26,23 @@ protected String getStackDescription(SpellAbility sa) {
     @Override
     public void resolve(SpellAbility sa) {
         final Card card = sa.getHostCard();
-        CardCollectionView list;
+        final Player activator = sa.getActivatingPlayer();
         final String valid = sa.getParamOrDefault("ValidCards", "");
+        Map<Player, CardCollection> untapMap = Maps.newHashMap();
 
-        if (!sa.usesTargeting() && !sa.hasParam("Defined")) {
-            list = sa.getActivatingPlayer().getGame().getCardsIn(ZoneType.Battlefield);
-        } else {
-            list = getTargetPlayers(sa).getCardsIn(ZoneType.Battlefield);
+        for (Player p : getTargetPlayers(sa)) {
```

Comment:
looks like this was already broken for _Reins of Power_ some time ago
Just noting it since fixing that probably depends on how this part ends up

Reply:
I think I have Reins of Power and friends all working properly now... I did include lines for `ControllerUntaps$` – is there a card that even needs that yet?

Reply:
I guess other than Early Harvest